plugin/ochttp/propagation/tracecontext/propagation.go

```diff
@@ -87,15 +90,71 @@ func (f *HTTPFormat) SpanContextFromRequest(req *http.Request) (sc trace.SpanCon
 		return trace.SpanContext{}, false
 	}
 
+	// Extract Tracestate
+	tracestate, ok := tracestateFromRequest(req)
+	if ok == false {
+		return trace.SpanContext{}, false
+	}
+	sc.Tracestate = tracestate
 	return sc, true
 }
 
-// SpanContextToRequest modifies the given request to include a header.
+func tracestateFromRequest(req *http.Request) (*tracestate.Tracestate, bool) {
+	h := req.Header.Get(tracestateHeader)
+	if h == "" {
+		return nil, true
+	}
+
+	var entries []tracestate.Entry
+	pairs := strings.Split(h, ",")
+	headerLenWithoutTrailingSpaces := len(pairs) - 1 // Number of commas
```

Comment:
var headerLen int
headerLen += len(pairs) - 1 // number of commas

What happens if len(pairs) is 0?